Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -673,4 +662,77 @@ bool abstract_eq(Interpreter& interpreter, Value lhs, Value rhs)
     return false;
 }
 
+TriState abstract_relation(Interpreter& interpreter, bool left_first, Value lhs, Value rhs)
+{
+    Value x_primitive = {};
+    Value y_primitive = {};
+
+    if (left_first) {
+        x_primitive = lhs.to_primitive(interpreter);
+        if (interpreter.exception())
+            return {};
+        y_primitive = rhs.to_primitive(interpreter);
+        if (interpreter.exception())
+            return {};
+    } else {
+        y_primitive = lhs.to_primitive(interpreter);
+        if (interpreter.exception())
+            return {};
+        x_primitive = rhs.to_primitive(interpreter);
+        if (interpreter.exception())
+            return {};
+    }
+
+    if (x_primitive.type() == Value::Type::String && y_primitive.type() == Value::Type::String) {
+        auto x_string = x_primitive.as_string().string();
+        auto y_string = y_primitive.as_string().string();
+
+        if (x_string.starts_with(y_string))
+            return TriState::False;
+        if (y_string.starts_with(x_string))
+            return TriState::True;
+
+        Utf8View x_codepoints { x_string };
+        Utf8View y_codepoints { y_string };
+        for (auto k = x_codepoints.begin(), l = y_codepoints.begin();
+             k != x_codepoints.end() && l != y_codepoints.end();
+             ++k, ++l) {
+            if (*k != *l) {
+                if (*k < *l) {
+                    return TriState::True;
+                } else {
+                    return TriState::False;
+                }
+            }
+        }
+        ASSERT_NOT_REACHED();
+    }
+
+    // FIXME add BigInt cases here once we have BigInt
+
+    Value x_numeric = x_primitive.to_number(interpreter);
```

Comment:
Not that it matters that much, but I'd say Serenity code should try to leverage `auto` as much as possible - 

```suggestion
    auto x_numeric = x_primitive.to_number(interpreter);
```

same for `double x_value = x_numeric.as_double();` - looks like Java :smile: